src/searchModal/advancedSearch.js

```diff
@@ -91,11 +91,12 @@ export default function advancedSearchFactory(config) {
         /**
          * Builds substring of search query with the advanced criteria conditions
          */
-        getAdvancedCriteriaQuery: function () {
+        getAdvancedCriteriaQuery: function (hasSearchInput) {
             const advancedSearchCriteria = _.filter(criteriaState, criterion => criterion.rendered === true);
             let query = '';
 
             advancedSearchCriteria.forEach(renderedCriterion => {
+                query += hasSearchInput || query.trim().length !== 0 ? " AND " : '';
```

Comment:
It makes no sense to add an empty string in case of condition returns false
```suggestion
                if (hasSearchInput || query.trim().length !== 0) {
                    query += ? " AND "
                }
```